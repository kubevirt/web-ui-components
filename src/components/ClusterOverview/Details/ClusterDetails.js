import React from 'react';
import PropTypes from 'prop-types';

import { ClusterOverviewContext } from '../ClusterOverviewContext';
import { getOpenshiftVersion, getClusterName, getInfrastructurePlatform } from '../../../selectors';
import { InlineLoading } from '../../Loading';
import { Details } from './Details';

export const ClusterDetails = ({ infrastructure, openshiftClusterVersions, LoadingComponent }) => {
  const items = {
    name: {
      title: 'Name',
      value: getClusterName(infrastructure),
      isLoading: !infrastructure,
    },
    provider: {
      title: 'Provider',
      value: getInfrastructurePlatform(infrastructure),
      isLoading: !infrastructure,
    },
    rhhi: {
      title: 'RHHI version',
      value: 'hardcoded-version', // this will be hardcoded for the demo
      isLoading: false,
    },
    openshift: {
      title: 'Openshift version',
      value: getOpenshiftVersion(openshiftClusterVersions),
      isLoading: !openshiftClusterVersions,
    },
  };

  return <Details items={items} LoadingComponent={LoadingComponent} />;
};

ClusterDetails.defaultProps = {
  infrastructure: null,
  openshiftClusterVersions: null,
  LoadingComponent: InlineLoading,
};

ClusterDetails.propTypes = {
  infrastructure: PropTypes.object,
  openshiftClusterVersions: PropTypes.array,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

export const ClusterDetailsConnected = () => (
  <ClusterOverviewContext.Consumer>{props => <ClusterDetails {...props} />}</ClusterOverviewContext.Consumer>
);
