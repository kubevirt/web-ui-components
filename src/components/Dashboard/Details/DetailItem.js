import React from 'react';
import PropTypes from 'prop-types';
import { OverlayTrigger, Tooltip } from 'patternfly-react';

import { DASHES } from '../../../constants';

export const DetailItem = ({ title, value, isLoading, LoadingComponent }) => {
  const description = value ? (
    <OverlayTrigger
      overlay={<Tooltip id={`tooltip-for-${title}`}>{value}</Tooltip>}
      placement="top"
      trigger={['hover', 'focus']}
      rootClose={false}
    >
      <span>{value}</span>
    </OverlayTrigger>
  ) : (
    DASHES
  );
  return (
    <React.Fragment>
      <dt className="kubevirt-detail__item-title">{title}</dt>
      <dd className="kubevirt-detail__item-value">{isLoading ? <LoadingComponent /> : description}</dd>
    </React.Fragment>
  );
};

DetailItem.defaultProps = {
  value: null,
};

DetailItem.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};
