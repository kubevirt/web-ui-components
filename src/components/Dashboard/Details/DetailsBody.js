import React from 'react';
import PropTypes from 'prop-types';

export const DetailsBody = ({ children }) => <dl>{children}</dl>;

DetailsBody.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};
