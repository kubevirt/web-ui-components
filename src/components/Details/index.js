export * from './VmDetails';
export * from './VmTemplateDetails';
export * from './BootOrder';
