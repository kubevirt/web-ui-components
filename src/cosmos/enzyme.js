import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
