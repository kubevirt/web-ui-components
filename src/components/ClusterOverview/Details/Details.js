import React from 'react';
import PropTypes from 'prop-types';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
} from '../../Dashboard/DashboardCard';
import { ClusterOverviewContext } from '../ClusterOverviewContext';
import { InlineLoading } from '../../Loading';
import { getClusterName, getInfrastructurePlatform, getOpenshiftVersion } from '../../../selectors';
import { DetailItem } from '../../Dashboard/Details/DetailItem';
import { DetailsBody } from '../../Dashboard/Details/DetailsBody';

export const Details = ({ infrastructure, openshiftClusterVersions, LoadingComponent }) => (
  <DashboardCard>
    <DashboardCardHeader>
      <DashboardCardTitle>Details</DashboardCardTitle>
    </DashboardCardHeader>
    <DashboardCardBody>
      <DetailsBody>
        <DetailItem
          key="name"
          title="Name"
          value={getClusterName(infrastructure)}
          isLoading={!infrastructure}
          LoadingComponent={LoadingComponent}
        />
        <DetailItem
          key="provider"
          title="Provider"
          value={getInfrastructurePlatform(infrastructure)}
          isLoading={!infrastructure}
          LoadingComponent={LoadingComponent}
        />
        <DetailItem
          key="rhhi"
          title="RHHI version"
          value="1.0" // this will be hardcoded for the demo
          isLoading={false}
          LoadingComponent={LoadingComponent}
        />
        <DetailItem
          key="openshift"
          title="Openshift version"
          value={getOpenshiftVersion(openshiftClusterVersions)}
          isLoading={!openshiftClusterVersions}
          LoadingComponent={LoadingComponent}
        />
      </DetailsBody>
    </DashboardCardBody>
  </DashboardCard>
);

Details.defaultProps = {
  infrastructure: null,
  openshiftClusterVersions: null,
  LoadingComponent: InlineLoading,
};

Details.propTypes = {
  infrastructure: PropTypes.object,
  openshiftClusterVersions: PropTypes.array,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

export const DetailsConnected = () => (
  <ClusterOverviewContext.Consumer>{props => <Details {...props} />}</ClusterOverviewContext.Consumer>
);
