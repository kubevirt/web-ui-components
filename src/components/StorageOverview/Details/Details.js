import React from 'react';
import PropTypes from 'prop-types';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
} from '../../Dashboard/DashboardCard';
import { StorageOverviewContext } from '../StorageOverviewContext';
import { InlineLoading } from '../../Loading';
import { DetailItem } from '../../Dashboard/Details/DetailItem';
import { DetailsBody } from '../../Dashboard/Details/DetailsBody';
import { getName } from '../../../selectors';

export const StorageDetails = ({ LoadingComponent, cephCluster, className }) => (
  <DashboardCard className={className}>
    <DashboardCardHeader>
      <DashboardCardTitle>OCS Details</DashboardCardTitle>
    </DashboardCardHeader>
    <DashboardCardBody>
      <DetailsBody>
        <DetailItem
          key="name"
          title="Name"
          value={cephCluster && cephCluster[0] && getName(cephCluster[0])}
          isLoading={!cephCluster}
          LoadingComponent={LoadingComponent}
        />
        <DetailItem
          key="provider"
          title="Provider"
          value="Bare Metal"
          isLoading={false}
          LoadingComponent={LoadingComponent}
        />
      </DetailsBody>
    </DashboardCardBody>
  </DashboardCard>
);

StorageDetails.defaultProps = {
  cephCluster: null,
  LoadingComponent: InlineLoading,
  className: null,
};

StorageDetails.propTypes = {
  cephCluster: PropTypes.array,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  className: PropTypes.string,
};

export const StorageDetailsConnected = ({ className }) => (
  <StorageOverviewContext.Consumer>
    {props => <StorageDetails {...props} className={className} />}
  </StorageOverviewContext.Consumer>
);

StorageDetailsConnected.defaultProps = {
  className: null,
};

StorageDetailsConnected.propTypes = {
  className: PropTypes.string,
};
