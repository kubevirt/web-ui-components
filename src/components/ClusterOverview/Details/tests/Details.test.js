import React from 'react';
import { render } from 'enzyme';

import { Details } from '../Details';
import { default as DetailsFixtures } from '../fixtures/Details.fixture';

const testDetailsOverview = () => <Details {...DetailsFixtures[0].props} />;

describe('<Details />', () => {
  it('renders correctly', () => {
    const component = render(testDetailsOverview());
    expect(component).toMatchSnapshot();
  });
});
