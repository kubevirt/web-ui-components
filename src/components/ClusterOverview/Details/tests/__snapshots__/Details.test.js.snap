// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> renders correctly 1`] = `
<div
  class="card-pf kubevirt-dashboard__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <h2
      class="card-pf-title"
    >
      Details
    </h2>
  </div>
  <div
    class="card-pf-body"
  >
    <dl>
      <dt
        class="kubevirt-detail__item-title"
      >
        Name
      </dt>
      <dd
        class="kubevirt-detail__item-value"
      >
        <span>
          clusterName.clusterDoman
        </span>
      </dd>
      <dt
        class="kubevirt-detail__item-title"
      >
        Provider
      </dt>
      <dd
        class="kubevirt-detail__item-value"
      >
        <span>
          AWS
        </span>
      </dd>
      <dt
        class="kubevirt-detail__item-title"
      >
        RHHI version
      </dt>
      <dd
        class="kubevirt-detail__item-value"
      >
        <span>
          1.0
        </span>
      </dd>
      <dt
        class="kubevirt-detail__item-title"
      >
        Openshift version
      </dt>
      <dd
        class="kubevirt-detail__item-value"
      >
        <span>
          v4.0.0
        </span>
      </dd>
    </dl>
  </div>
</div>
`;
